blueretro_presets.js

```diff
@@ -341,41 +371,46 @@ function writeInputCfg(cfgId, cfg) {
 }
 
 function saveInput() {
-    document.getElementById("inputSaveText").style.display = 'none';
+    //get consoleName and preset name
     var preset = Number(document.getElementById("presetsName").value);
-    var nbMapping = presets[preset].map.length;
-    var cfgSize = nbMapping*8 + 3;
-    var cfg = new Uint8Array(cfgSize);
-    var cfgId = Number(document.getElementById("inputSelect").value);
+    var consoleName = Number(document.getElementById("consoleName").value);
+    //make sure both console and preset are not placeholders before we do anything
+    if(preset != -1 && consoleName != -1){
```

Comment:
Didn't had time to run this locally to see, but are the selection box empty now when page load?

Reply:
first box is output defaulting to Output 1, second box says select a console, third box was left blank until a console was selected. on second thought i feel having the third box say "select a preset" as place holder until a console is selected looks better. no saving can be done until both a console and a preset is selected.

![image](https://user-images.githubusercontent.com/52015201/160909316-cd8b4103-3c66-41e1-a9e3-0f283494f3ff.png)

changed to

![image](https://user-images.githubusercontent.com/52015201/160910899-d66e2e38-9543-4747-a195-24f19c8597d9.png)

the selection box text can be changed trivially. let me know what you prefer.

Reply:
I think 2nd one is better, also maybe let people skip console filter and make all map available in the preset list initially.